plugin/types.go

```diff
@@ -63,9 +63,14 @@ type Root interface {
 }
 
 // ExecOptions is a struct we can add new features to that must be serializable to JSON.
-// Examples of potential features: user, privileged, tty, map of environment variables, string of stdin, timeout.
+// Examples of potential features: user, privileged, map of environment variables, timeout.
 type ExecOptions struct {
+	// Stdin can be used to pass
 	Stdin io.Reader
+	// Tty instructs the execution environment to allocate a TTY. In some cases this changes the
+	// behavior of the command being executed. For commands that don't naturally exit, it also
+	// instructs the execution environment to ensure the command is terminated when the context ends.
+	Tty bool
```

Comment:
Hmm, how about something like:

> Tty instructs the executor to allocate a TTY (pseudo-terminal), which lets Wash communicate with the running process via its Stdin. The TTY is used to send a process termination signal (Ctrl+C) via Stdin when the passed-in Exec context is cancelled.
> 
> NOTE TO PLUGIN AUTHORS: The Tty option is only relevant for executors that do not have an API endpoint to stop a running command (e.g. Docker, Kubernetes). If your executor does have an API endpoint to stop a running command, then ignore the Tty option. Note that the reason we make Tty an option instead of having the relevant executors always attach a TTY is because attaching a TTY can change the behavior of the command that's being executed.
>
> NOTE TO CALLERS: The Tty option is useful for executing your own stream-like commands (e.g. `tail -f`), because it ensures that there are no orphaned processes after the request is cancelled/finished.
>

^ This wording is basically a compilation of all your answers to my questions about this, which I imagine are questions that other plugin authors/callers might have when they see this option.

Reply:
Done.